hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -626,16 +626,15 @@ private void addVolume(final StorageLocation location, final String storageId,
     StorageType storageType = location.getStorageType();
     final FsVolumeImpl fsVolume =
         createFsVolume(sd.getStorageUuid(), sd, config);
-    VolumeReplicaMap tempVolumeMap = new VolumeReplicaMap(
-        new AutoCloseableLock());
+    VolumeReplicaMap tempVolumeMap = null;
     ArrayList<IOException> exceptions = Lists.newArrayList();
 
     for (final NamespaceInfo nsInfo : nsInfos) {
       String bpid = nsInfo.getBlockPoolID();
       try {
         fsVolume.addBlockPool(bpid, config, this.timer);
-        tempVolumeMap.addAll(
-            fsVolume.getVolumeMap(bpid, fsVolume, ramDiskReplicaTracker));
+        tempVolumeMap =
```

Comment:
Only those in the last nsInfo will go into the tempVolumeMap?

Reply:
Earlier, `addAll` was adding all the `k-v` in the `ProvidedVolumeReplicaMap` into the  `VolumeReplicaMap` (`tempVolumeMap `).  Which means the `replicaMap` associated with the `ProvidedVolume` will still be of type `VolumeReplicaMap` and not `ProvidedVolumeReplicaMap`. The `ProvidedVolume` will never invoke `aliasMap` request if it is not connected to `ProvidedVolumeReplicaMap`.

Reply:
I meant the semantics associated with the {{for (final NamespaceInfo nsInfo : nsInfos) }} is not the same as before right?

Reply:
Good catch. I will fix this this. Thanks !